spec/menubar/utils_spec.lua

```diff
@@ -65,6 +65,65 @@ describe("menubar.utils lookup_icon_uncached", function()
         assert_found_in_path('awesome',  '/.icons/awesome/64x64/apps/awesome.png')
         assert_found_in_path('awesome2', '/.icons/awesome/scalable/apps/awesome2.png')
     end)
+
+-- Check for no argument :: if not icon_file or icon_file == "" then
+    it('no icon specified', function()
+        assert.is_false(utils.lookup_icon_uncached())
+        assert.is_false(utils.lookup_icon_uncached(nil)) -- Same as above?
+        assert.is_false(utils.lookup_icon_uncached(false))
+        assert.is_false(utils.lookup_icon_uncached(''))
+    end)
+
+-- Check for icon not in search path :: if icon_file:sub(1, 1) == '/' and supported_icon_formats[icon_file_ext] then
+    it('finds icons even those not in the search paths when full path specified', function()
+
+        -- Shimmed icon base directories contain the following icons:
+        --
+        --     usr/share/icon5.png
+        --     usr/share/icon6.xpm
+        --     usr/share/icon7.svg
+
+        assert_found_in_path(root..'/usr/share/icon5.png', '/usr/share/icon5.png')
```

Comment:
and if this test is passing for you locally?

Reply:
Thanks for trying, but we must be missing something else

Local test is not through Travis but on a local build...
```
local utils = require("menubar.utils")
print("1>",utils.lookup_icon_uncached('/usr/share/icon5.png'))
print("2>",utils.lookup_icon_uncached('/usr/share/icon6.xpm'))
print("3>",utils.lookup_icon_uncached('/usr/share/icon7.svg'))

1>	/usr/share/icon5.png
2>	/usr/share/icon6.xpm
3>	/usr/share/icon7.svg
```

Could I make a suggestion on the PR https://github.com/awesomeWM/awesome/pull/2542 ?
- Cancel the change request and approve
- Raise a new issue to get the commented test to work [and assign it to me if you want]
- Would that work for you?

Reply:
but do you have locally the actual file `/usr/share/icon5.png`?

Reply:
and no, first fix your test, next pr will be merged

Reply:
that's the point, first you need to remove it from your local. and next makes tests pass without those files in /usr/share but in root../usr/share

Reply:
Done - No change - local branch builds perfectly

Reply:
what do you mean, did you removed from your /usr/share those icons now and tests are passing locally?

Reply:
I see lots of tests running and passing when it builds - What exactly am I looking for or supposed to run?

Reply:
install `busted` and run inside git repo root

Reply:
Same as on github

Not getting anything from lookup_icon_uncached

No change if I put a file in my /usr/share

Reply:
ok, so first thing what you don't need to put that file there

second: you need to modify testcode to make it work locally, as a start -- you could add few print statements to your function on important steps to see why it's not returning anything

Reply:
Original shimming was not sufficient and needed rework